projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobDeploymentCrudAsyncIT.java

```diff
@@ -24,6 +29,61 @@
     webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
     classes = ControlplaneApplication.class)
 public class DataJobDeploymentCrudAsyncIT extends BaseDataJobDeploymentCrudIT {
+
+  @Autowired private DesiredJobDeploymentRepository desiredJobDeploymentRepository;
+  @Autowired private ActualJobDeploymentRepository actualJobDeploymentRepository;
+
+  @Override
+  protected void beforeDeploymentDeletion() {
+    var desiredDataJobDeployment = desiredJobDeploymentRepository.findById(testJobName);
+    var actualDataJobDeployment = actualJobDeploymentRepository.findById(testJobName);
+
+    checkDesiredDeployment(desiredDataJobDeployment);
+    checkActualDeployment(actualDataJobDeployment);
+  }
+
   @Override
-  protected void beforeDeploymentDeletion() {}
+  protected void afterDeploymentDeletion() {
+    var desiredDataJobDeployment = desiredJobDeploymentRepository.findById(testJobName);
+    var actualDataJobDeployment = actualJobDeploymentRepository.findById(testJobName);
+
+    checkDesiredDeploymentDeleted(desiredDataJobDeployment);
+    checkActualDeploymentDeleted(actualDataJobDeployment);
+  }
+
+  private void checkDesiredDeployment(Optional<DesiredDataJobDeployment> desiredDataJobDeployment) {
```

Comment:
If it is possible the test shouldn't use repositories and services but only the controllers/APIs.